### PR TITLE
Fix/176 label mapping

### PR DESCRIPTION
**Pull request by zohrehasadi00**
## Requesting Review

Tag the people who should review this:
@Tenebrae311 @fhilgers @Paul2607 @saribx @ChristophMantsch @BenediktZinn @anuunchin @zohrehasadi00

Map analyzer detection class IDs to human-readable labels in backend metadata.
Expose COCO label lookup via shared data module.

### CODE REVIEW

File: src/backend/tests/analyzer/test_manager.py [hunk 1]
```diff
@@ -158,3 +158,23 @@ async def test_process_detection_excludes_updated_tracks_from_interpolation(
         manager._tracking_manager.get_interpolated_detections_and_distances.call_args
     )
     assert call_args[1]["track_ids_to_exclude"] == {0}
+
+
+def test_build_metadata_message_maps_label_to_text(
```
Comment (anuunchin) [inline src/backend/tests/analyzer/test_manager.py]: Might be a good idea to parametrize this to test for edge cases like unknown labels

File: src/backend/common/data/__init__.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# SPDX-FileCopyrightText: 2025 robot-visual-perception
```
Comment (anuunchin) [inline src/backend/common/data/__init__.py]: I'm not sure if we need these here, since it seems we're only importing from the submodule directly `from common.data.coco_labels import get_coco_label`

Comment (anuunchin) [reply src/backend/common/data/__init__.py]: F401 this is a good indicator we don't need this

File: src/backend/analyzer/manager.py [hunk 3]
```diff
@@ -407,7 +408,7 @@ def _build_metadata_message(
                         "width": norm_w,
                         "height": norm_h,
                     },
-                    "label": det.cls_id,
+                    "label": get_coco_label(det.cls_id),
```
Comment (anuunchin) [inline src/backend/analyzer/manager.py]: I think we need adjustments here or/and in the frontend, since frontend is still expecting int labels

Comment (anuunchin) [reply src/backend/analyzer/manager.py]: hmm, bad that the tests aren't catching this cause we have no e2e tests yet

**Comment by zohrehasadi00** [comment]
> One important thing only -> mismatch in what the frontend expects 🧠

Lovely comments. I resolved them, but please have another look too. Tnx.
